### PR TITLE
combining button primary and button secondary

### LINKED ISSUE #555: Refactor Button Components
## Description

In Lib/components all re-usable components are collected which are used in both Site and App.

Since this PR https://github.com/KlimaDAO/klimadao/pull/550 a new Button component was introduced "ButtonSecondary"
which copy pasted the same styles again from ButtonPrimary.

In order to not maintain too many styles, let´s simplify the integration of the re-usable buttons.

- [x] import same styles from ButtonPrimary to ButtonSecondary and adjust only the styles which are different (button border and background)
- [ ] combine both component in a better way (maybe let ButtonSecondary render Buttonprimary and just pass styles)
- [x] add the same variants to ButtonSecondary "gray" | "icon" | "blueRounded" 
- [x] ensure all buttons are still working on both App and Site

## Button Primary and Secondary variant "blueRounded"
Bildschirmfoto 2022-07-29 um 14 52 25

## Button Primary default and with variant="gray" in lightMode
Bildschirmfoto 2022-07-29 um 14 54 02

## Button Primary default and with variant="gray" in darkMode
Bildschirmfoto 2022-07-29 um 14 54 15

## Button Primary default
Bildschirmfoto 2022-07-29 um 14 53 13
Bildschirmfoto 2022-07-29 um 14 52 48

### PR DESCRIPTION
**Pull request by goatsOnDrugs**
## Description
I combined the primary and secondary button component because thare are almost the same. Secondary is a boolean you can pass to button primary but i dont know if "button primary" is a good name if its also secondary button. Should we still have separate components? 

## Related Ticket
#555 

Resolves #555 

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: lib/components/Buttons/ButtonPrimary.tsx [hunk 1]
```diff
@@ -14,26 +14,43 @@ interface Props {
...
+      props.secondary ? styles.button_secondary : styles.button_primary,
+      props.className,
+      {
+        icon: props.variant === "icon",
+      }
     );
-  } else {
-    buttonStyle = cx(styles.button_primary, props.className, {
-      icon: props.variant === "icon",
-    });
```
Comment (ladytrekker) [inline lib/components/Buttons/ButtonPrimary.tsx:36]: This whole block is very hard to read.

I would write the styles like this:

```TS
  const buttonStyle = cx(
    styles.buttonPrimary,
    {
      gray: props.variant === "gray",
    },
    {
      blueRounded: props.variant === "blueRounded",
    },
    {
      icon: props.variant === "icon",
    },
    props.className
  );
  ```

Comment (ladytrekker) [reply lib/components/Buttons/ButtonPrimary.tsx:36]: And in the styles for `buttonPrimary`

```TS
export const buttonPrimary = css`
  ${buttonBase};
  background-color: var(--klima-green);

  &:active:not(:disabled) {
    transform: scale(0.95);
    opacity: 0.6;
  }
  &,
  &:hover:not(:disabled),
  &:visited {
    color: white; /* same in darkmode */
  }

  &:disabled {
    background-color: var(--surface-03);
    cursor: not-allowed;
  }

  &.icon {
    gap: 0.8rem;
  }

  &.gray {
    background-color: var(--surface-01);
    color: var(--font-02);

    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--font-02); /* same in darkmode */
    }
  }

  &.blueRounded {
    background-color: var(--klima-blue);
    color: var(--surface-01);
    border-radius: 0.8rem;
    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--surface-01); /* same in darkmode */
    }
  }
`;
```

File: lib/components/Buttons/ButtonSecondary.tsx [hunk 2]
```diff
@@ -1,65 +0,0 @@
...
-    <button
-      type="button"
-      className={buttonStyle}
-      onClick={props.onClick}
-      disabled={props.disabled}
-    >
-      {props.label}
-    </button>
-  );
-};
```
Comment (ladytrekker) [inline lib/components/Buttons/ButtonSecondary.tsx:65]: I would keep the component `ButtonSecondary` 

and create a BaseButton Component for both `ButtonPrimary` and `ButtonSecondary` which pass their styles to this component.


So like


```TS
// BaseButton
export interface Props {
  label: ReactNode;
  className?: string;
  onClick?: () => void;
  href?: string;
  variant?: "gray" | "icon" | "blueRounded" | null;
  link?: Link;
  rel?: string;
  target?: string;
  isExternalHref?: boolean;
  disabled?: boolean;
}

interface BaseProps extends Props {
  buttonStyle: string;
}

export const BaseButton: FC<BaseProps> = ({ link: LinkElement, ...props }) => {
  if (props.href) {
    if (LinkElement) {
      return (
        <LinkElement href={props.href}>
          <a className={props.buttonStyle}>{props.label}</a>
        </LinkElement>
      );
    }

    return (
      <a {...props} className={props.buttonStyle} href={props.href}>
        {props.label}
      </a>
    );
  }

  return (
    <button
      type="button"
      className={props.buttonStyle}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.label}
    </button>
  );
};
```

Comment (ladytrekker) [reply lib/components/Buttons/ButtonSecondary.tsx:65]: Button primary

```TS

import { BaseButton, Props } from "./BaseButton";


export const ButtonPrimary: FC<Props> = (props) => {
  const buttonStyle = cx(
    styles.buttonPrimary,
    {
      gray: props.variant === "gray",
    },
    {
      blueRounded: props.variant === "blueRounded",
    },
    {
      icon: props.variant === "icon",
    },
    props.className
  );

  return <BaseButton {...props} buttonStyle={buttonStyle} />;
};
```

Comment (ladytrekker) [reply lib/components/Buttons/ButtonSecondary.tsx:65]: Button Secondary

```TS

import { BaseButton, Props } from "./BaseButton";

export const ButtonSecondary: FC<Props> = (props) => {
  const buttonStyle = cx(
    styles.buttonSecondary,
    {
      gray: props.variant === "gray",
    },
    {
      blueRounded: props.variant === "blueRounded",
    },
    {
      icon: props.variant === "icon",
    },
    props.className
  );
  return <BaseButton {...props} buttonStyle={buttonStyle} />;
};
```

File: lib/components/Buttons/styles.ts [hunk 3]
```diff
@@ -37,38 +37,15 @@ export const button_primary = css`
...
   }
-
-  &:disabled {
-    border-color: var(--surface-03);
-    cursor: not-allowed;
-  }
-
-  &.icon {
-    gap: 0.8rem;
-  }
```
Comment (ladytrekker) [inline lib/components/Buttons/styles.ts:71]: In this file I would make a clear distinction between base, Primary and Secondary

```TS
const buttonBase = css`
  ${button};
  display: flex;
  padding: 0rem 2.4rem;
  align-items: center;
  justify-content: center;
  appearance: none;
  text-decoration: none;
  /* min-height to conform with Lighthouse min tap-target */
  min-height: 4.8rem;
  cursor: pointer;
  border-radius: 0.4rem;
  transition: opacity 0.3s ease 0s;
`;

export const buttonPrimary = css`
  ${buttonBase};
  background-color: var(--klima-green);

  &:active:not(:disabled) {
    transform: scale(0.95);
    opacity: 0.6;
  }
  &,
  &:hover:not(:disabled),
  &:visited {
    color: white; /* same in darkmode */
  }

  &:disabled {
    background-color: var(--surface-03);
    cursor: not-allowed;
  }

  &.icon {
    gap: 0.8rem;
  }

  &.gray {
    background-color: var(--surface-01);
    color: var(--font-02);

    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--font-02); /* same in darkmode */
    }
  }

  &.blueRounded {
    background-color: var(--klima-blue);
    color: var(--surface-01);
    border-radius: 0.8rem;
    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--surface-01); /* same in darkmode */
    }
  }
`;
export const buttonSecondary = css`
  ${buttonBase}
  border: 1px solid var(--klima-green);
  border-radius: 0.4rem;
  background-color: transparent;
  &,
  &:hover:not(:disabled),
  &:visited {
    color: var(--klima-green); /* same in darkmode */
  }

  &.gray {
    border-color: var(--surface-01);
    color: var(--surface-01);

    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--font-01); /* same in darkmode */
    }
  }

  &.blueRounded {
    border-color: var(--klima-blue);
    color: var(--klima-blue);
    border-radius: 0.8rem;
    &,
    &:hover,
    &:hover:not(:disabled),
    &:visited {
      color: var(--klima-blue); /* same in darkmode */
    }
  }
`;
```

**Review by ladytrekker — COMMENTED** [review]
Thanks.

I made a few suggestions for keeping both components and simplify their rendering.

Let me know what you think.

File: lib/components/Buttons/ButtonPrimary.tsx (hunk 1 above)
Comment (Atmosfearful) [reply lib/components/Buttons/ButtonPrimary.tsx:36]: @ladytrekker @goatsOnDrugs lady is right we can avoid javascript and rely on cx, but actually you can condense it into one object.
```
 const buttonStyle = cx(
    styles.buttonPrimary,
    {
      gray: props.variant === "gray",
      blueRounded: props.variant === "blueRounded",
      icon: props.variant === "icon",
    },
    props.className
  );
```

**Comment by goatsOnDrugs** [comment]
@ladytrekker thanks for reviewing this. Great suggestions! I implemented everything from the comments above.

**Comment by goatsOnDrugs** [comment]
@Atmosfearful I think this one is good to go if yall dont have any more feedback.

**Review by ladytrekker — APPROVED** [review]
Thanks, now it's more precise

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**